### PR TITLE
#5924: Enable unet on wormhole_b0 changes

### PR DESCRIPTION
**Pull request by tapspatel**

### CODE REVIEW

**Comment by tapspatel** [comment]
Issue: https://github.com/tenstorrent-metal/tt-metal/issues/5924#issuecomment-1981469484
Post-commit pipeline: https://github.com/tenstorrent-metal/tt-metal/actions/runs/8216981925

File: tests/ttnn/unit_tests/operations/test_conv2d.py [hunk 1]
```diff
@@ -969,3 +971,104 @@ def test_unet_conv(
...
+    use_1d_systolic_array,
+    config_override,
+    use_shallow_conv_variant,
+    output_layout,
+):
+    if output_layout == ttnn.ROW_MAJOR_LAYOUT and activations_dtype == ttnn.bfloat8_b:
+        pytest.skip("Row major layout not compatible with bfloat8_b")
+    if output_layout == ttnn.ROW_MAJOR_LAYOUT and input_height >= 1056:
+        pytest.skip("OOM")
+    breakpoint()
```
Comment (mywoodstock) [inline tests/ttnn/unit_tests/operations/test_conv2d.py:1052]: Intended?

File: tests/ttnn/unit_tests/operations/test_conv2d.py [hunk 2]
```diff
@@ -969,3 +971,104 @@ def test_unet_conv(
...
+        ),  # fails. mismatch. It passes when input_channels=64. Probably an issue with padding when input_channels % 32 != 0.
+        (2, 16, 16, 528, 80, 3, 3, 1, 1, 1, 1, True, None, False),
+        (2, 16, 32, 1056, 160, 3, 3, 1, 1, 1, 1, True, {"act_block_h": 5 * 32}, False),
+        (2, 16, 16, 1056, 160, 3, 3, 1, 1, 1, 1, True, {"act_block_h": 5 * 32}, False),
+        (2, 1, 16, 1056, 160, 3, 3, 1, 1, 1, 1, True, {"act_block_h": 5 * 32}, False),
+    ),
+)
+@pytest.mark.parametrize(
+    "weights_dtype",
+    [ttnn.bfloat8_b],
```
Comment (mywoodstock) [inline tests/ttnn/unit_tests/operations/test_conv2d.py:1018]: Testing for bfloat16 weights?

**Review by mywoodstock — APPROVED** [review]

File: ttnn/ttnn/operations/others.py [hunk 3]
```diff
@@ -325,7 +326,7 @@ def _get_upsample_shard_grid_from_num_shards(ncores: int):
 ## helper function for upsample
 def _get_upsample_num_shards(batch_size: int, height: int, num_channels: int, shard_strategy: ttnn.ShardStrategy):
     ## calculate ncores, corresponding grid_size and in_shard_shape based on the input_shape
-    max_grid_size = (9, 12)  ## (y, x)
+    max_grid_size = (9, 12) if os.getenv("ARCH_NAME", "grayskull") == "grayskull" else (8, 8)  ## (y, x)
```
[the remark this answers is not in the document]
Comment (tapspatel) [reply ttnn/ttnn/operations/others.py:329]: I don't have access to the device pointer in this part of the stack trace

File: tests/ttnn/unit_tests/operations/test_conv2d.py (hunk 1 above)
Comment (tapspatel) [reply tests/ttnn/unit_tests/operations/test_conv2d.py:1052]: removed

File: tests/ttnn/unit_tests/operations/test_conv2d.py (hunk 2 above)
Comment (tapspatel) [reply tests/ttnn/unit_tests/operations/test_conv2d.py:1018]: added

File: ttnn/ttnn/operations/others.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (eyonland) [reply ttnn/ttnn/operations/others.py:329]: @cfjchu , should this information be coming from your Multi Device data structures?

**Review by tt-nshanker — APPROVED** [review]

File: ttnn/ttnn/operations/others.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cfjchu) [reply ttnn/ttnn/operations/others.py:329]: This isn't specific to multi-device, but that's a great point, users shouldn't have to infer grid-size like this. I'll add some changes to make this information about core-grid more readily available.

**Review by cfjchu — APPROVED** [review]

File: ttnn/ttnn/operations/others.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tapspatel) [reply ttnn/ttnn/operations/others.py:329]: Filed an issue for that support: https://github.com/tenstorrent-metal/tt-metal/issues/6308

[the remark this answers is not in the document]
Comment (arakhmati) [reply ttnn/ttnn/operations/others.py:329]: @tapspatel you can pass in `device` into `_get_upsample_num_shards`and `_get_upsample_shard_grid_from_num_shards` from `input_tensor.device()`

**Review by arakhmati — CHANGES_REQUESTED** [review]

**Review by arakhmati — APPROVED** [review]

**Pull request merged**